src/cache/collection_ttl.rs

```diff
@@ -0,0 +1,92 @@
+use std::time::Duration;
+
+/// Represents the desired behavior for managing the TTL on collection objects.
+///
+/// For cache operations that modify a collection (dictionaries, lists, or sets), there
+/// are a few things to consider. The first time the collection is created, we need to
+/// set a TTL on it. For subsequent operations that modify the collection you may choose
+/// to update the TTL in order to prolong the life of the cached collection object, or
+/// you may choose to leave the TTL unmodified in order to ensure that the collection
+/// expires at the original TTL.
+///
+/// The default behaviour is to refresh the TTL (to prolong the life of the collection)
+/// each time it is written using the client's default item TTL.
+#[derive(Copy, Clone, Debug)]
+pub struct CollectionTtl {
+    ttl: Option<Duration>,
+    refresh: bool,
+}
+
+impl CollectionTtl {
```

Comment:
while you are in here can you do a quick spot check on the dev docs / javascript version of this class and add any convenience factory fns that we are missing? e.g. I am pretty sure we document `CollectionTtl.of(...)` as the most common factory fn in JS.